infra/openai.tf

```diff
@@ -1,38 +1,54 @@
+resource "random_integer" "this" {
+  max = 999999
+  min = 100000
+}
+
 module "openai" {
-  source              = "Azure/openai/azurerm"
-  version             = "0.1.3"
+  source              = "Azure/avm-res-cognitiveservices-account/azurerm"
+  version             = "0.4.0"
+  kind                = "OpenAI"
+  name                = "azure-openai-${random_integer.this.result}"
```

Comment:
Can we use the `naming` module?


Example:
https://github.com/Azure/terraform-azurerm-avm-ptn-aks-production/blob/207ab0de42b47ef4ea4772bcefebcd3bf4c8304f/examples/default/main.tf#L39-L42

https://github.com/Azure/terraform-azurerm-avm-ptn-aks-production/blob/207ab0de42b47ef4ea4772bcefebcd3bf4c8304f/examples/default/main.tf#L47

the module has support for cognitive account:
https://github.com/Azure/terraform-azurerm-naming/blob/9aaa2b0f58383501a4025a4696e4225504ffa9c3/main.tf#L339-L348